tests/retry_logic_simple_test.py

```diff
@@ -0,0 +1,61 @@
+import pytest
+from unittest.mock import Mock, patch
+
+from traveltimepy import Client, AsyncClient
+from traveltimepy.errors import TravelTimeJsonError, TravelTimeServerError
+
+
+class TestRetryLogic:
+    @pytest.mark.asyncio
+    async def test_async_server_error_retries(self):
+        client = AsyncClient("test", "test", retry_attempts=2)
+
+        async def mock_retry_method(*args, **kwargs):
+            raise TravelTimeServerError("Server error")
+
+        with patch.object(client, "_make_request", side_effect=mock_retry_method):
+            with pytest.raises(
+                Exception
+            ):  # Could be RetryError or TravelTimeServerError
+                await client._make_request("GET", "https://test.com", {}, Mock)
+
+    def test_sync_server_error_retries(self):
+        client = Client("test", "test", retry_attempts=2)
```

Comment:
We shall call the close() method at the end or use "with" clause - even if just as an example of proper usage